vowpalwabbit/core/include/vw/core/global_data.h

```diff
@@ -116,6 +116,7 @@ class workspace
   std::unique_ptr<all_reduce_base> all_reduce;
 
   bool chain_hash_json = false;
+  bool calculate_feature_space_hash = false;

```

Comment:
Do we know by how much we improve performance by doing this in setup_example? Perhaps we should avoid adding to the global state until a point where it is proven to be a performance issue?

Reply:
did a quick benchmark, the difference isn't really noticeable so moved calculation into LAS